src/Caching/Memory/test/CapacityTests.cs

```diff
@@ -155,10 +157,10 @@ public async Task ExceedsCapacityCompacts()
             });
 
             var entryOptions = new MemoryCacheEntryOptions { Size = 6 };
-            var sem = new SemaphoreSlim(0, 1);
+            var tcs = new TaskCompletionSource<object>();
```

Comment:
TCS needs TaskCreationOptions.RunContinuationsAsynchronously to actually help here.

Reply:
Ah, now I recall. I was looking into this test failure before my vacation and it seemed more like a threadpool starvation issue now that more tests are run in parallel. I added logging to confirm this but I just realized the build wasn't capturing the functional logs. I've fixed that in https://github.com/aspnet/Extensions/pull/799. I'll wait for the logs to confirm.